### ISSUE TITLE
Can't download the blockchain / HDD is read-only

### ISSUE THREAD

**Issue by szche** [raspiblitz/raspiblitz#233]
I've been trying to download the blockchain for hours now.
The torrent is stuck at "Inactive:", and FTP fails do download right after I turn it on.
At first, I thought it's because of my internet connection, but pinging google works just fine.
Then, I found out that when I try to nano a txt file in the HDD directory (/mnt/hdd/), I can't save the file because I get "Read-only file system" error. After trying to make it writable I get "write-protected" error.
Anyone got any ideas what's wrong?

**Comment by fluidvoice** [comment]
what is your out put of... `ls -al /mnt/hdd`
try downloading with this:
`sudo wget -c -r -P /mnt/hdd/download -q --show-progress ftp://f00f39c4:download@w0189aba.kasserver.com/`

**Comment by szche** [comment]
> what is your out put of... `ls -al /mnt/hdd`

```
total 24
drwxr-xr-x 3 root root  4096 Jan  1 10:10 .
drwxr-xr-x 3 root root  4096 Jan  1 10:11 ..
drwx------ 2 root root 16384 Jan  1 10:10 lost+found
```


> try downloading with this:
> `sudo wget -c -r -P /mnt/hdd/download -q --show-progress ftp://f00f39c4:download@w0189aba.kasserver.com/`

When I try this, nothing happens

**Comment by fluidvoice** [comment]
try:
`sudo mkdir /mnt/hdd/download`
then try that wget command again.
if it "does nothing" what do you see on the screen/terminal?
are you able to at least ping that machine that is the FTP server?...
`ping w0189aba.kasserver.com`

**Comment by fluidvoice** [comment]
What happens when you run this?...
`./00mainMenu.sh`

**Comment by szche** [comment]
> try:
> `mkdir /mnt/hdd/download`

`mkdir: cannot create directory ‘/mnt/hdd/download’: Read-only file system`

![screenshot_5](https://user-images.githubusercontent.com/43548343/50572923-0cc06100-0dca-11e9-9b2d-97194689420b.png)


And yes, pinging works perfectly fine

**Comment by szche** [comment]
> What happens when you run this?...
> `./00mainMenu.sh`

I get the setup window (Choose either Bitcoin or Litecoin setup)


Here's the response during HDD init if you're interested
![screenshot_4](https://user-images.githubusercontent.com/43548343/50572938-7b052380-0dca-11e9-81a1-a31cc1ed9745.png)

**Comment by fluidvoice** [comment]
That looks OK so far. What happens later?

**Comment by szche** [comment]
I have to choose how to download the copy of blockchain

When I choose Torrent here's the process:
![screenshot_3](https://user-images.githubusercontent.com/43548343/50572992-55c4e500-0dcb-11e9-8ca4-d88e13f2a084.png)

And then it's stuck at "Inactive"
![screenshot_6](https://user-images.githubusercontent.com/43548343/50572995-67a68800-0dcb-11e9-83fc-f857c1a0e820.png)

And when I choose FTP I get this screen almost instantly (there's a short print showing the link to the ftp server)
![screenshot_7](https://user-images.githubusercontent.com/43548343/50573009-c79d2e80-0dcb-11e9-98c8-8a043a420c6e.png)






Can't get past this point

**Comment by fluidvoice** [comment]
are you able to at least ping that machine that is the FTP server?...
`ping w0189aba.kasserver.com`

**Comment by szche** [comment]
> 
> 
> are you able to at least ping that machine that is the FTP server?...
> `ping w0189aba.kasserver.com`

pinging works perfectly fine
At this point I'm pretty certain the issue is that the HDD is in read-only mode

**Comment by fluidvoice** [comment]
I was looking at the script. Strange how it does this with a background process. What's your output of:
`sudo screen -S Download -ls`

**Comment by fluidvoice** [comment]
do either of these succeed or same read-only error?
```
sudo -u root mkdir /mnt/hdd/download
sudo -u pi mkdir /mnt/hdd/download
```

**Comment by szche** [comment]
>  `sudo screen -S Download -ls`

`No Sockets found in /run/screen/S-root.`

> sudo -u root mkdir /mnt/hdd/download
> sudo -u pi mkdir /mnt/hdd/download

Both of these give me the same error
`mkdir: cannot create directory ‘/mnt/hdd/download’: Read-only file system`

**Comment by fluidvoice** [comment]
may be a clue?
https://www.raspberrypi.org/forums/viewtopic.php?t=210349#p1299075
as per this thread, you see any errors or pertinent things (like USB drive coming on/off line) from a grep like this, or similar?...
`sudo grep -i usb /var/log/syslog`

**Comment by fluidvoice** [comment]
Do you have another HDD that you can try and see if the same problem happens? I wonder if even a thumb/pen drive could be used just as a test - or if the setup scripts will detect it's not big enough for the ~230GB blockchain and so disallow it.

**Comment by szche** [comment]
There seems to be some errors there.
Unfortunately don't have any other drives to test it.
If the power usage is the problem, how do I solve it then?

**Comment by fluidvoice** [comment]
You can also try these manual steps to see if it gives insights to the problem:

verify your drive is /dev/sda with `fdisk -l`
try remounting it as read-write:
`sudo mount -o remount,rw /dev/sda`
if it complains about write-protected, you may also have to run `sudo hdparm -r0 /dev/sda`
Then try remounting read-write again.
if success, then try creating that dir again:
`sudo mkdir /mnt/hdd/download`

**Comment by fluidvoice** [comment]
> There seems to be some errors there.
> Unfortunately don't have any other drives to test it.
> If the power usage is the problem, how do I solve it then?

Is it a 2.5" drive? mounted in an *un-powered* external case?
One possibility is the un-powered case is sucking too much power from the PI, so perhaps a different case will work, or using a powered USB case.  Also... just thought... have you tried plugging the drive into a different one of the 4 Pi USB ports? Maybe one is flaky. Try a different USB cable as well.  Is it a short or long cable?
Another thing just occurred to me. What is the watt/amps rating of your Pi power supply?  Perhaps it's too low or poor quality and this is causing issues driving your USB HDD.

**Comment by fluidvoice** [comment]
Please post the syslog errors you see, as this could help other people who might have the same problem as you.

**Comment by szche** [comment]
fdisk returns this:
```
Disk /dev/sda: 931.5 GiB, 1000204886016 bytes, 1953525168 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xfb09f60f
```

But when I try to `sudo mount -o remount,rw /dev/sda` I get `mount: /dev/sda not mounted or bad option`

**Comment by szche** [comment]
I wiped and reflashed the sd card so I can't get the full log, but here's the current log 
```
Jan  1 14:33:29 RaspiBlitz kernel: [  255.207488] sd 0:0:0:0: [sda] tag#0 UNKNOWN(0x2003) Result: hostbyte=0x00 driverbyte=0x08
Jan  1 14:33:29 RaspiBlitz kernel: [  255.207500] sd 0:0:0:0: [sda] tag#0 Sense Key : 0x5 [current]
Jan  1 14:33:29 RaspiBlitz kernel: [  255.207506] sd 0:0:0:0: [sda] tag#0 ASC=0x24 ASCQ=0x0
Jan  1 14:33:29 RaspiBlitz kernel: [  255.207515] sd 0:0:0:0: [sda] tag#0 CDB: opcode=0x2a 2a 08 00 00 08 00 00 00 08 00
Jan  1 14:33:29 RaspiBlitz kernel: [  255.207521] print_req_error: critical target error, dev sda, sector 2048
Jan  1 14:33:29 RaspiBlitz kernel: [  255.207527] Buffer I/O error on dev sda1, logical block 0, lost sync page write
Jan  1 14:33:29 RaspiBlitz kernel: [  255.349616] EXT4-fs (sda1): mounted filesystem with ordered data mode. Opts: (null)
Jan  1 14:33:35 RaspiBlitz _background.sh[593]: auto-unlock is OFF
Jan  1 14:33:37 RaspiBlitz kernel: [  263.634476] sd 0:0:0:0: [sda] tag#0 UNKNOWN(0x2003) Result: hostbyte=0x00 driverbyte=0x08
Jan  1 14:33:37 RaspiBlitz kernel: [  263.634492] sd 0:0:0:0: [sda] tag#0 Sense Key : 0x5 [current]
Jan  1 14:33:37 RaspiBlitz kernel: [  263.634504] sd 0:0:0:0: [sda] tag#0 ASC=0x24 ASCQ=0x0
Jan  1 14:33:37 RaspiBlitz kernel: [  263.634519] sd 0:0:0:0: [sda] tag#0 CDB: opcode=0x2a 2a 08 3a 04 08 18 00 00 08 00
Jan  1 14:33:37 RaspiBlitz kernel: [  263.634530] print_req_error: critical target error, dev sda, sector 973342744
Jan  1 14:33:37 RaspiBlitz kernel: [  263.634544] print_req_error: critical target error, dev sda, sector 973342744
Jan  1 14:33:37 RaspiBlitz kernel: [  263.634575] Aborting journal on device sda1-8.
Jan  1 14:33:37 RaspiBlitz kernel: [  263.635185] sd 0:0:0:0: [sda] tag#0 UNKNOWN(0x2003) Result: hostbyte=0x00 driverbyte=0x08
Jan  1 14:33:37 RaspiBlitz kernel: [  263.635197] sd 0:0:0:0: [sda] tag#0 Sense Key : 0x5 [current]
Jan  1 14:33:37 RaspiBlitz kernel: [  263.635209] sd 0:0:0:0: [sda] tag#0 ASC=0x24 ASCQ=0x0
Jan  1 14:33:37 RaspiBlitz kernel: [  263.635221] sd 0:0:0:0: [sda] tag#0 CDB: opcode=0x2a 2a 08 3a 04 08 00 00 00 08 00
Jan  1 14:33:37 RaspiBlitz kernel: [  263.635229] print_req_error: critical target error, dev sda, sector 973342720
Jan  1 14:33:37 RaspiBlitz kernel: [  263.635241] Buffer I/O error on dev sda1, logical block 121667584, lost sync page write
Jan  1 14:33:37 RaspiBlitz kernel: [  263.635272] JBD2: Error -5 detected when updating journal superblock for sda1-8.
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914030] EXT4-fs (sda1): previous I/O error to superblock detected
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914764] sd 0:0:0:0: [sda] tag#0 UNKNOWN(0x2003) Result: hostbyte=0x00 driverbyte=0x08
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914782] sd 0:0:0:0: [sda] tag#0 Sense Key : 0x5 [current]
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914794] sd 0:0:0:0: [sda] tag#0 ASC=0x24 ASCQ=0x0
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914809] sd 0:0:0:0: [sda] tag#0 CDB: opcode=0x2a 2a 08 00 00 08 00 00 00 08 00
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914818] print_req_error: critical target error, dev sda, sector 2048
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914837] Buffer I/O error on dev sda1, logical block 0, lost sync page write
Jan  1 14:33:38 RaspiBlitz kernel: [  263.914889] EXT4-fs error (device sda1): ext4_journal_check_start:61: Detected aborted journal
Jan  1 14:33:38 RaspiBlitz kernel: [  263.925198] EXT4-fs (sda1): Remounting filesystem read-only
Jan  1 14:33:38 RaspiBlitz kernel: [  263.932401] EXT4-fs (sda1): previous I/O error to superblock detected
Jan  1 14:33:38 RaspiBlitz kernel: [  263.933125] sd 0:0:0:0: [sda] tag#0 UNKNOWN(0x2003) Result: hostbyte=0x00 driverbyte=0x08
Jan  1 14:33:38 RaspiBlitz kernel: [  263.933147] sd 0:0:0:0: [sda] tag#0 Sense Key : 0x5 [current]
Jan  1 14:33:38 RaspiBlitz kernel: [  263.933158] sd 0:0:0:0: [sda] tag#0 ASC=0x24 ASCQ=0x0
Jan  1 14:33:38 RaspiBlitz kernel: [  263.933174] sd 0:0:0:0: [sda] tag#0 CDB: opcode=0x2a 2a 08 00 00 08 00 00 00 08 00
Jan  1 14:33:38 RaspiBlitz kernel: [  263.933183] print_req_error: critical target error, dev sda, sector 2048
Jan  1 14:33:38 RaspiBlitz kernel: [  263.933197] Buffer I/O error on dev sda1, logical block 0, lost sync page write

```

This one line is interesting
`Jan  1 14:33:38 RaspiBlitz kernel: [  263.925198] EXT4-fs (sda1): Remounting filesystem read-only`

**Comment by fluidvoice** [comment]
If you're sure the HDD is not defective, or flaky, I'm guessing the problem is power related.  Perhaps using an AC powered USB hub between the HDD and Pi will help. Or getting a better Pi power supply.  You tried already swapping USB ports and USB cables, right?

**Comment by szche** [comment]
After checking my drawer I found another HDD that after connecting to Raspberry started working.
At first, I thought it was the drive's fault, so I swapped old drive to a new case. Surprisingly it stopped working after that.  Finally, I swapped the cases and now it's working flawlessly.
Sorry for wasting so much of your time @fluidvoice and I really appreciate your help.

TL;DR: The problem was with faulty HDD case

**Comment by fluidvoice** [comment]
Nice! glad you got it working.  It's fine. In helping people I'm learning a lot about how all these shell scripts work. Sadly, I'm not impressed with them. They are very fragile things and hard to troubleshoot when something goes wrong.
